Modules/_dbmmodule.c

```diff
@@ -76,16 +84,25 @@ newdbmobject(const char *file, int flags, int mode)
 static void
 dbm_dealloc(dbmobject *dp)
 {
-    if ( dp->di_dbm )
+    if (dp->di_dbm) {
         dbm_close(dp->di_dbm);
-    PyObject_Del(dp);
+    }
+    PyTypeObject *tp = Py_TYPE(dp);
+    tp->tp_free(dp);
+    Py_DECREF(tp);
 }
 
 static Py_ssize_t
 dbm_length(dbmobject *dp)
 {
+    PyTypeObject *tp = Py_TYPE(dp);
+    _dbm_state *state =  PyType_GetModuleState(tp);
+    assert(state != NULL);
+    if (state == NULL) {
+        return -1;
+    }
```

Comment:
Well, I propose to replace these checks with `assert(state != NULL);` for all PyType_GetModuleState() calls, not only in this function.